Cargo.toml

```diff
@@ -86,6 +86,9 @@ full = [
 http1 = []
 http2 = ["h2"]
 
+# Support for non-canonical HTTP/1 reason phrases
+http1_reason_phrase = ["http1"]
```

Comment:
Yea, so let's talk about if we need a feature. My initial gut feeling is that it seems like such a small feature that it doesn't pull its weight, just include it. Especially if the benchmarks don't show a noticeable difference.

Reply:
I'd be in favor of including it. I think it'd take some very unusual traffic (like maxing out reason phrase length on every response) to cause a noticeable difference. It also helps that the cost is incurred on responses rather than on requests which are usually more concerning as a DoS vector.

I'll go ahead and fold it into `http1` but will keep it in a separate commit while this is under review.

Reply:
Made this change in d621c741, which led to a pleasant reduction in the amount of awkwardly `cfg`ed code :sparkles: